### PR TITLE
JNI bindings for NTH_ELEMENT window aggregation

### PR DESCRIPTION
**Pull request by mythrocks**
Depends on #11158.

This commit adds the JNI bindings required to invoke `NTH_ELEMENT` aggregations as a window aggregation. Java tests are included.

### CODE REVIEW

**Comment by mythrocks** [comment]
Note: While the changes listed include the changes from #11158, only the changes under the `java/` directory are material to this review.

File: java/src/test/java/ai/rapids/cudf/TableTest.java [hunk 1]
```diff
@@ -5315,6 +5315,55 @@ void testWindowingMean() {
...
 
+  @Test
+  void testWindowingNthElement() {
+    final Integer X = null;
+    try (Table unsorted = new Table.TestBuilder().column( 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1) // GBY Key
+        .column( 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3) // GBY Key
+        .column( 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6) // OBY Key
+        .column( 7, 5, 3, 7, 7, 9, X, 4, X, 0, 4, X) // Agg Column
+        .build()) {
+      try (Table sorted = unsorted.orderBy(OrderByArg.asc(0), OrderByArg.asc(1), OrderByArg.asc(2));
```
Comment (razajafri) [inline java/src/test/java/ai/rapids/cudf/TableTest.java:5326]: May be a more meaningful test if the table `unsorted` wasn't already sorted?

**Comment by mythrocks** [comment]
Setting `DO NOT MERGE` until the dependency is merged.

File: java/src/test/java/ai/rapids/cudf/TableTest.java [hunk 2]
```diff
@@ -5315,6 +5315,55 @@ void testWindowingMean() {
     }
   }
 
+  @Test
+  void testWindowingNthElement() {
+    final Integer X = null;
+    try (Table unsorted = new Table.TestBuilder().column( 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1) // GBY Key
```
Comment (rwlee) [inline java/src/test/java/ai/rapids/cudf/TableTest.java:5321]:
```suggestion
    try (Table unsorted = new Table.TestBuilder().column(1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1) // GBY Key
```

**Review by rwlee — COMMENTED** [review]
LGTM pending dependency being merged. 1 formatting/whitespace nit

**Comment by mythrocks** [comment]
@razajafri, @rwlee, I have addressed your concerns. I'd appreciate it if you could take another gander.
Since #11158 has been merged, rebasing this change has reduced the diff to its core.

**Review by rwlee — APPROVED** [review]

**Review by razajafri — APPROVED** [review]

**Comment by mythrocks** [comment]
Thank you for the reviews, @rwlee, @razajafri.

**Pull request merged**